doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -433,9 +462,10 @@ Creating Hints
          hint_info ::= %| {? @natural } {? @one_pattern }
          one_pattern ::= @one_term
 
-      The first form adds each :n:`@qualid` as a hint with the head symbol of the type of
-      :n:`@qualid` to the specified hint databases (:n:`@ident`\s). The cost of the hint is the number of
-      subgoals generated by :tacn:`simple apply` :n:`@qualid` or, if specified, :n:`@natural`. The
+      The first form adds each :n:`@qualid` as a hint with the head symbol of the
+      type of :n:`@qualid` to the specified hint databases (:n:`@ident`\s). The
+      cost of the hint is the number of subgoals generated by :tacn:`simple apply`
+      :n:`@qualid` or, if specified, :n:`@natural`. The
```

Comment:
I take whatever Print HintDb prints as accurate.  It is printing the actual tactic that will be executed.  So I think you're mistaken.  Feel free to prove me wrong.

"The cost of the hint is the number of subgoals generated" - that doesn't make much sense either because the costs are just constants in the hint and they can't depend on the result of applying the hint.  Unless tc eauto is actually adding up the costs.  I'll check that again.

Reply:
simple apply is different from what the internals actually use. As already mentioned, this is why we have autoapply. The debug printer is just that, a debug printer. Any similarity with reality would be fortuitous.

Reply:
Indeed.  We should disclose the difference in the tactics under `Print HintDb`.  @Janno if can you tell me what the best equivalents are, I will update `Hints.pr_hint`.  The 3 *auto tactics refer to slightly different code, so harder to tell if they are identical across the three.

How often do the best equivalents fail?  If it's more than a little, we should make the exact tactics available in Gallina.

Reply:
@ppedrot is more of an expert here as far as the implementation is concerned. I use `autoapply @hint with the_hint_db` to mimic the application of the hint. But AFAIK that is not exactly the tactic that is run by the OCaml code. I don't understand the differences, though.